dojo/problem/redis.py

```diff
@@ -0,0 +1,155 @@
+import base64
+import json
+import logging
+from dataclasses import dataclass, field
+from functools import lru_cache
+
+import redis
+
+import dojo.problem.helper as problems_help
+from dojo.models import Finding
+
+logger = logging.getLogger(__name__)
+
+SEVERITY_ORDER = {
+    "Critical": 5,
+    "High": 4,
+    "Medium": 3,
+    "Low": 2,
+    "Info": 1,
+}
+
+
+@dataclass
+class Problem:
+    name: str
+    problem_id: str
+    severity: str
+    script_ids: set = field(default_factory=set)
+    finding_ids: set = field(default_factory=set)
+
+    def to_dict(self):
+        return {
+            "name": self.name,
+            "problem_id": self.problem_id,
+            "severity": self.severity,
+            "script_ids": list(self.script_ids),
+            "finding_ids": list(self.finding_ids),
+        }
+
+    @staticmethod
+    def from_dict(data):
+        return Problem(
+            name=data["name"],
+            problem_id=data["problem_id"],
+            severity=data["severity"],
+            script_ids=set(data.get("script_ids", [])),
+            finding_ids=set(data.get("finding_ids", [])),
+        )
+
+
+@lru_cache(maxsize=1)
+def get_redis_client():
+    return redis.Redis(host="django-defectdojo-redis-1", port=6379, decode_responses=True)
+
+
+def add_finding_to_redis(finding):
+    if finding.vuln_id_from_tool and finding.severity != "Info":
+        redis_client = get_redis_client()
+        if redis_client.exists("problems", "id_to_problem"):
+            script_id = finding.vuln_id_from_tool
+            script_to_problem_mapping = problems_help.load_json()
+            problem_id = script_to_problem_mapping.get(script_id, script_id)
+            problem_id = base64.b64encode(problem_id.encode()).decode()
```

Comment:
This code appears somewhere else in this file... we should not duplicate code like this.  It should be encapsulated in a function and be implemented only once.